### ISSUE TITLE
Add Typescript Type definitions

### ISSUE THREAD

**Issue by Clumsy-Coder** [zackurben/alphavantage#82]
Type definitions for each of the functions available.

Data structure returned when calling the functions.
Data structure when fetched data is polished `alpha.util.polish`

**Comment by Clumsy-Coder** [comment]
I have a few worked out

### Raw Data from AlphaVantage API
```ts
// https://www.alphavantage.co/documentation/#latestprice
interface RawStockQuote {
    'Global Quote': {
      '01. symbol': string;
      '02. open': string;
      '03. high': string;
      '04. low': string;
      '05. price': string;
      '06. volume': string;
      '07. latest trading day': string;
      '08. previous close': string;
      '09. change': string;
      '10. change percent': string;
    };
  }
```

```ts
// https://www.alphavantage.co/documentation/#dailyadj
interface RawStockDailyAdjusted {
    'Meta Data': {
      '1. Information': string;
      '2. Symbol': string;
      '3. Last Refreshed': string;
      '4. Output Size': string;
      '5. Time Zone': string;
    };
    'Time Series (Daily)': {
      [date: string]: {
        '1. open': string;
        '2. high': string;
        '3. low': string;
        '4. close': string;
        '5. adjusted close': string;
        '6. volume': string;
        '7. dividend amount': string;
        '8. split coefficient': string;
      };
    };
  }
```

```ts
// https://www.alphavantage.co/documentation/#symbolsearch
interface RawStockSearch {
    bestMatches: {
      '1. symbol': string;
      '2. name': string;
      '3. type': string;
      '4. region': string;
      '5. marketOpen': string;
      '6. marketClose': string;
      '7. timezone': string;
      '8. currency': string;
      '9. matchScore': string;
    }[];
  }
```

### Polished Data### 

```ts
// https://www.alphavantage.co/documentation/#latestprice
interface StockQuote {
    data: {
      symbol: string;
      open: string;
      high: string;
      low: string;
      price: string;
      volume: string;
      latest_trading_day: string;
      prev_close: string;
      change: string;
      change_percent: string;
    };
  }
```

```ts
// https://www.alphavantage.co/documentation/#dailyadj
interface StockDailyAdjusted {
    meta: {
      information: string;
      symbol: string;
      updated: string;
      size: string;
      zone: string;
    };
    data: {
      [date: string]: {
        open: string;
        high: string;
        low: string;
        close: string;
        adjusted: string;
        volume: string;
        dividend: string;
        split: string;
      };
    };
  }
```

```ts
// https://www.alphavantage.co/documentation/#symbolsearch
interface StockSearch {
    bestMatches: {
      symbol: string;
      name: string;
      type: string;
      region: string;
      market_open: string;
      market_close: string;
      zone: string;
      currency: string;
      match_score: string;
    }[];
  }
```

**Comment by zackurben** [comment]
This was addressed in 2.1.0, thanks.
